bob-apps/bin/bobp.rs

```diff
@@ -505,13 +513,32 @@ async fn get_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statis
         keys.shuffle(&mut thread_rng());
         Box::new(keys.into_iter())
     };
+
+    let do_basic_auth = task_conf.basic_username.is_some() && task_conf.basic_password.is_some();
```

Comment:
The logic of adding auth to request is the same for both workers, and should be extracted.
One possible way this can be achieved is by adding function `get_request_creator`, which will create closure `Fn(T) -> Request<T>` producing requests and adding auth if needed.